mp/src/game/client/momentum/ui/leaderboards/LeaderboardsTimes.cpp

```diff
@@ -529,21 +557,20 @@ void CLeaderboardsTimes::OnlineTimesVectorToLeaderboards(TimeType_t type)
     }
 }
 
-bool CLeaderboardsTimes::GetPlayerTimes(KeyValues* outPlayerInfo, bool fullUpdate)
+bool CLeaderboardsTimes::FillPlayerTimes(bool fullUpdate)
 {
-    if (!outPlayerInfo)
-        return false;
+    if (!m_bTimesNeedUpdate[TIMES_LOCAL])
+    {
+        float lastUp = gpGlobals->curtime - m_flTimesLastUpdate[TIMES_LOCAL];
+        m_bTimesNeedUpdate[TIMES_LOCAL] = fullUpdate && lastUp >= UPDATE_INTERVAL;
```

Comment:
Shouldn't this be `||`?

Reply:
Wait nevermind, that would cause this to reread all the runs every `UPDATE_INTERVAL` seconds